databricks/koalas/indexes.py

```diff
@@ -831,6 +831,50 @@ def name(self) -> str:
     def name(self, name: str) -> None:
         raise PandasNotImplementedError(class_name='pd.MultiIndex', property_name='name')
 
+    def swaplevel(self, i=-2, j=-1):
+        """
+        Swap level i with level j.
+        Calling this method does not change the ordering of the values.
+
+        Parameters
+        ----------
+        i : int, str, default -2
+            First level of index to be swapped. Can pass level name as string.
+            Type of parameters can be mixed.
+        j : int, str, default -1
+            Second level of index to be swapped. Can pass level name as string.
+            Type of parameters can be mixed.
+
+        Returns
+        -------
+        MultiIndex
+            A new MultiIndex.
+
+        Examples
+        --------
+        >>> midx = ks.MultiIndex.from_arrays([['a', 'b'], [1, 2]], names = ['word', 'number'])
+        >>> midx
+        MultiIndex([('a', 1),
+                    ('b', 2)],
+                   names=['word', 'number'])
+
+        >>> midx.swaplevel(0, 1)
+        MultiIndex([(1, 'a'),
+                    (2, 'b')],
+                   names=['number', 'word'])
+
+        >>> midx.swaplevel('number', 'word')
+        MultiIndex([(1, 'a'),
+                    (2, 'b')],
+                   names=['number', 'word'])
+        """
+        i = i if isinstance(i, int) else self.names.index(i)
+        j = j if isinstance(j, int) else self.names.index(j)
+        index_map = self._internal.index_map
+        index_map[i], index_map[j], = index_map[j], index_map[i]
```

Comment:
i think this line overwrite existing `index_map`.

below example may can help you understand this situation.

```python
>>> a = [1, 2, 3]
>>> b = a
>>> b[0], b[1] = b[1], b[0]
>>> b
[2, 1, 3]
>>> a
[2, 1, 3]
```

Reply:
Thanks for your kind explain. I got it.

Reply:
my pleasure :)

Reply:
Do we need  rename spark columns name '__index_level_n__' when level change?

Reply:
well... i think maybe that's not necessary since the only factor we need manage to implement this is `index_map`.

but actually i have not enough insight of `_InternalFrame`, so it should better ask to someone other have enough insight of this.

maybe can @ueshin , @HyukjinKwon help us? 😸

Reply:
Yes, waiting for help.

Reply:
That should be okay with not updating the column names.